enterprise/suppress-user/internal/badgerdb/badgerdb.go

```diff
@@ -146,6 +151,72 @@ func (b *Repository) Suppressed(workspaceID, userID, sourceID string) (bool, err
 	return true, nil
 }
 
+func (b *Repository) GetCreatedAt(workspaceID, userID, sourceID string) (time.Time, error) {
```

Comment:
Instead of going through all this trouble, we may simply modify the Suppressed method signature to return a pointer to suppression metadata if it exists.

Note: backwards compatibility should be carefully thought out